pkg/operation/botanist/component/extensions/operatingsystemconfig/original/components/kernelconfig/component.go

```diff
@@ -33,7 +38,21 @@ func (component) Name() string {
 	return "kernel-config"
 }
 
-func (component) Config(_ components.Context) ([]extensionsv1alpha1.Unit, []extensionsv1alpha1.File, error) {
+func (component) Config(ctx components.Context) ([]extensionsv1alpha1.Unit, []extensionsv1alpha1.File, error) {
+	var newData = data
+	if (ctx.KubeletConfigParameters.ProtectKernelDefaults != nil && *ctx.KubeletConfigParameters.ProtectKernelDefaults) ||
+		(ctx.KubeletConfigParameters.ProtectKernelDefaults == nil && ctx.KubernetesVersion != nil && version.ConstraintK8sGreaterEqual126.Check(ctx.KubernetesVersion)) {
+		newData += "#Needed configuration by kubelet\n" +
+			"#The kubelet sets these values but it is not able to when protectKernelDefaults=true\n" +
+			"#Ref https://github.com/gardener/gardener/issues/7069\n" +
+			fmt.Sprintf("%s = %s\n", sysctl.VMOvercommitMemory, strconv.Itoa(sysctl.VMOvercommitMemoryAlways)) +
+			fmt.Sprintf("%s = %s\n", sysctl.VMPanicOnOOM, strconv.Itoa(sysctl.VMPanicOnOOMInvokeOOMKiller)) +
+			fmt.Sprintf("%s = %s\n", sysctl.KernelPanicOnOops, strconv.Itoa(sysctl.KernelPanicOnOopsAlways)) +
+			fmt.Sprintf("%s = %s\n", sysctl.KernelPanic, strconv.Itoa(sysctl.KernelPanicRebootTimeout)) +
+			fmt.Sprintf("%s = %s\n", sysctl.RootMaxKeys, strconv.Itoa(sysctl.RootMaxKeysSetting)) +
+			fmt.Sprintf("%s = %s\n", sysctl.RootMaxBytes, strconv.Itoa(sysctl.RootMaxBytesSetting))
```

Comment:
I assume that these parameters depend or can depend on the use Kubernetes version?! Is there a reference we can cross check in the form of a unit test? Ideally, the unit test fails as soon as something has changed from Kubernetes's side.

Reply:
This is also what I think. Maybe we can have a test that will have the current values hardcoded and if something changes in k/k for example in `v1.26.0` this will start to fail when we revendor the new version.

Reply:
Nice catch! I will create such a test.